KillerApp/src/main/java/com/example/killerapp/Manager.java

```diff
@@ -0,0 +1,154 @@
+package com.example.killerapp;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.util.Log;
+
+import com.dezen.riccardo.smshandler.ReceivedMessageListener;
+import com.dezen.riccardo.smshandler.SMSManager;
+import com.dezen.riccardo.smshandler.SMSMessage;
+import com.dezen.riccardo.smshandler.SMSPeer;
+
+/***
+ * @author Pardeep
+ * Class
+ */
+
+public class Manager {
+    private static Context currentContext;
```

Comment:
I suggest using WeakReference<context>, it's a Class that allows the Context to be destroyed even if it is referenced here, it's useful to avoid memory leaks

Reply:
It's not showing it in the comment for some reason but such class should take Context as generic type so WeakReference\<Context\>

Reply:
ok i think i understood the concept , so i just need to add under Context the line
WeakReference<Context> weakReferance=new WeakReference<Context>(currentContext);  ?
i'm not sure of how i should use it